common/lib/src/realm_types.dart

```diff
@@ -79,6 +79,36 @@ class RealmStateError extends StateError implements RealmError {
   RealmStateError(String message) : super(message);
 }
 
+/// Thrown or reporeted if an error occurs during synchronization
+/// {@category Sync}
+class SyncError extends RealmError {
```

Comment:
Could we rename SyncError  to SyncErrorCode? upload_completion and  download_completion are working with realm_sync_error_code. We need to have another class SyncError for sync_error_handler argument.